src/invoice.rs

```diff
@@ -133,16 +159,21 @@ impl std::fmt::Display for RgbInvoice {
             write!(f, "{amt}+")?;
         }
         write!(f, "{}", self.beneficiary)?;
-        if !self.unknown_query.is_empty() {
+        if self.expiry.is_some() || !self.unknown_query.is_empty() {
             f.write_str("?")?;
         }
-        for (key, val) in self.unknown_query.iter().take(1) {
-            // TODO: URLEncode
-            write!(f, "{key}={val}")?;
+        let mut query_empty = true;
+        if let Some(expiry) = self.expiry {
+            write!(f, "expiry={}", expiry)?;
+            query_empty = false;
         }
-        for (key, val) in self.unknown_query.iter().skip(1) {
+        for (key, val) in self.unknown_query.iter() {
             // TODO: URLEncode
-            write!(f, "&{key}={val}")?;
+            if !query_empty {
+                f.write_str("&")?;
+            };
+            write!(f, "{key}={val}")?;
+            query_empty = false;
```

Comment:
I think the best way is to add `fn query_params(&self) -> IndexMap<String, String>` method to the Invoice and call it from here - leaving the previous code in place.